src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/DateTimeValueResolver.php

```diff
@@ -45,12 +51,18 @@ public function resolve(Request $request, ArgumentMetadata $argument): array
         $value = $request->attributes->get($argument->getName());
         $class = \DateTimeInterface::class === $argument->getType() ? \DateTimeImmutable::class : $argument->getType();
 
-        if ($value instanceof \DateTimeInterface) {
-            return [$value instanceof $class ? $value : $class::createFromInterface($value)];
+        if (!$value) {
+            if ($argument->isNullable()) {
+                return [null];
+            }
+            if (!$this->clock) {
+                return [new $class];
+            }
+            $value = $this->clock->now();
```

Comment:
By default, the `clock` service is a `NativeClock` instance with the default timezone. This is backward compatible with `new \DateTime()`.